File: C++/A/22A-secondorderstats.cpp
#include <bits/stdc++.h>
using namespace std;

class Solver
{
    int n;
    set<int> values;

public:
    void input()
    {
        cin >> n;
        for (int i = 0; i < n; i++)
        {
            int x;
            cin >> x;
            values.insert(x);
        }
    }
    string answer() { return ((values.size() != 1) ? to_string(*(++values.begin())) : "NO"); }
};

int main()
{
    Solver solution;
    solution.input();
    cout << solution.answer();
}

File: Python/A/22A-secondorderstats.py
class Solver:
    def __init__(self):
        self.n = 0
        self.values = set()

    def input(self):
        self.n = int(input())
        self.values = set(map(int, input().split()))
        self.values = sorted(self.values)

    def answer(self):
        return "NO" if len(self.values) == 1 else list(self.values)[1]


solution = Solver()
solution.input()
print(solution.answer())
